include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+     * This class was not designed for inheritance, and must not be subclassed.
+     *
+     * @see @ref coord::Coord
+     */
+    // TODO: add final keyword after migrating to Pybind11
+    class SphPoint
+    {
+    public:
+        /**
+         * @brief Represent a specific pair of spherical coordinates.
+         *
+         * @param longitude The azimuth angle of the point.
+         * @param latitude The elevation angle of the point. Must correspond
+         *                 to an angle in [-&pi;/2, &pi;/2] radians.
+         *
+         * @throws InvalidParameterError Thrown if any argument is NaN or
+         *                               infinite.
+         * @throws OutOfRangeError Thrown if @c latitude is out of range.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(Angle const & longitude, Angle const & latitude);
+
+        /**
+         * @brief Represent a specific direction in 3-dimensional space.
+         *
+         * The adopted convention is that <1, 0, 0> corresponds to (0, 0), that
+         * <0, 1, 0> corresponds to (&pi;/2, 0), and that <0, 0, 1> corresponds
+         * to (0, &pi;/2).
+         *
+         * @param vector A position whose direction will be stored as a point.
+         *               Must not be the zero vector. Need not be normalized.
+         *
+         * @throws InvalidParameterError Thrown if any element of @c vector is
+         *                               NaN or infinite, or if @c vector is
+         *                               the zero vector.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        explicit SphPoint(Point3D const & vector);
+
+        /**
+         * @brief Creates a copy of a SphPoint.
+         *
+         * @param other The point to copy.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(SphPoint const & other) = default;
+
+        /**
+         * Overwrites this object with the value of another SphPoint.
+         *
+         * This is the only method that can alter the state of a SphPoint after
+         * its construction, and is provided to allow in-place replacement of
+         * SphPoints where swapping is not possible.
+         *
+         * @param other The object with which to overwrite this one.
+         * @return a reference to this object.
+         *
+         * @par Exception Safety:
+         * This operator shall not throw exceptions.
+         */
+        SphPoint& operator=(SphPoint const & other) = default;
+
+        /*
+         * Accessors
+         */
+
+        /**
+         * @brief The azimuth angle of this point.
+         *
+         * If this point is at a coordinate pole, this method may return any
+         * value.
```

Comment:
Good. In thinking about it further I don't fully understand what you are warning about; I think it need some clarification. If I construct a SphPoint at the pole from a longitude and latitude then getLatitude reliably returns the original latitude. Furthermore, I think methods such as `rotated` will have a predictable result on the longitude (I'm not so sure about `offset`).

On the other hand, if a `SphPoint` is constructed from a Vector3D then `longitude` is not self-evident, though you can and probably ought to document what it will be. Similarly, if a method such as `offset` or `rotated` returns a point at the pole then longitude is probably not predictable.

So the situation is nuanced. It seems only a few methods can make a point with unpredictable longitude, and perhaps the warning about longitude belongs with those methods? If so, I'm not convinced anything need be said here (or can be said here without some additional explanation).